### ISSUE TITLE
Phi3.5-vision-instruct fine-tuning best practices. (Latex OCR Fine-tuning)

### ISSUE THREAD

**Issue by Jintao-Huang** [modelscope/ms-swift#1809]
Huggingface Model: https://huggingface.co/microsoft/Phi-3.5-vision-instruct

Fine-tuned Dataset: https://huggingface.co/datasets/linxy/LaTeX_OCR

Usually, fine-tuning a multimodal large model involves using a custom dataset for fine-tuning. Here, we will demonstrate a runnable demo.

Before starting the fine-tuning, please ensure that your environment is properly prepared.

```bash
git clone https://github.com/modelscope/ms-swift.git
cd swift
pip install -e .[llm]
```

### Inference

```bash
# ModelScope
CUDA_VISIBLE_DEVICES=0 swift infer \
  --model_type phi3_5-vision-instruct \
  --use_flash_attn false

# HuggingFace
USE_HF=1 CUDA_VISIBLE_DEVICES=0 swift infer \
  --model_type phi3_5-vision-instruct \
  --model_id_or_path microsoft/Phi-3.5-vision-instruct \
  --use_flash_attn false
```

**Results**
```
<<< who are you
I am Phi, an AI developed by Microsoft to assist with providing information, answering questions, and helping users find solutions to their queries. How can I assist you today?
--------------------------------------------------
<<< <image>please describe the image.
Input an image path or URL <<< http://modelscope-open.oss-cn-hangzhou.aliyuncs.com/images/cat.png
The image features a close-up of a kitten with striking blue eyes and a white and grey striped coat. The kitten's fur is soft and fluffy, and it appears to be looking directly at the camera with a curious and innocent expression. The background is blurred, which puts the focus entirely on the kitten's face.
--------------------------------------------------
<<<  <image>What is the result of the calculation?
Input an image path or URL <<< http://modelscope-open.oss-cn-hangzhou.aliyuncs.com/images/math.png
The result of the calculation 1452 + 45304 is 46756.
```

**GPU Memory**:

<img width="585" alt="截屏2024-08-23 18 09 29" src="https://github.com/user-attachments/assets/faee9bdd-1a39-4ca8-9258-5e43948b8cd4">

**Comment by Jintao-Huang** [comment]
### Fine-tuning

The format of the **custom dataset** is as follows (single image, multiple images, and no image):
```jsonl
{"query": "<image>55555", "response": "66666", "images": ["image_path"]}
{"query": "eeeee<image>eeeee<image>eeeee", "response": "fffff", "history": [], "images": ["image_path1", "image_path2"]}
{"query": "EEEEE", "response": "FFFFF", "history": [["query1", "response2"], ["query2", "response2"]], "images": []}
```

**Fine-tuning script**:
```bash
# HuggingFace
USE_HF=1 CUDA_VISIBLE_DEVICES=0,1,2,3 NPROC_PER_NODE=4 swift sft \
  --model_type phi3_5-vision-instruct \
  --model_id_or_path microsoft/Phi-3.5-vision-instruct \
  --sft_type lora \
  --dataset latex-ocr-print#20000 \
  --deepspeed default-zero2 \
  --output_dir output \
  --num_train_epochs 5 \
  --use_flash_attn false

# ModelScope
CUDA_VISIBLE_DEVICES=0,1,2,3 NPROC_PER_NODE=4 swift sft \
  --model_type phi3_5-vision-instruct \
  --sft_type lora \
  --dataset latex-ocr-print#20000 \
  --deepspeed default-zero2 \
  --output_dir output \
  --num_train_epochs 5 \
  --use_flash_attn false
```
If you want to use a custom dataset, simply specify as follows:

```bash
  --dataset train.jsonl \
  --val_dataset val.jsonl \
```

One of the data samples:

<img width="759" alt="截屏2024-08-23 19 15 50" src="https://github.com/user-attachments/assets/fa2bd49c-0dbd-4a13-8f98-2838a381c1be">

Number of trainable parameters：

<img width="678" alt="截屏2024-08-23 19 18 34" src="https://github.com/user-attachments/assets/963a89b8-7b2c-4b5c-b1db-02b8a73cad2d">

**GPU Memory**

<img width="580" alt="截屏2024-08-23 18 57 02" src="https://github.com/user-attachments/assets/81b32d28-b7af-4043-a673-9454dfeb0a3c">

**Training process**

<img width="751" alt="截屏2024-08-23 19 04 36" src="https://github.com/user-attachments/assets/277b2e17-54e4-4a2e-adb8-a97b654778c7">

**Train Loss** (Due to time constraints, we only fine-tuned for **1000 steps**)：

![train_loss (29)](https://github.com/user-attachments/assets/878d703f-fce4-4fed-b474-aaf065012258)

Here is the inference script after fine-tuning, we perform inference on the automatically segmented validation set:
```
# To run a full test, please set: `--show_dataset_sample -1`
# inference only
CUDA_VISIBLE_DEVICES=0 swift infer \
    --ckpt_dir output/phi3_5-vision-instruct/vx-xxx/checkpoint-xxx \
    --load_dataset_config true --use_flash_attn false

# merge-lora & inference
CUDA_VISIBLE_DEVICES=0 swift infer \
    --ckpt_dir output/phi3_5-vision-instruct/vx-xxx/checkpoint-xxx \
    --load_dataset_config true --merge_lora true \
    --safe_serialization false --use_flash_attn false
```

The results of the fine-tuned model inferring on the validation set: (Due to time constraints, we only fine-tuned for **1000 steps**)：

<img width="880" alt="截屏2024-08-23 19 34 01" src="https://github.com/user-attachments/assets/21cb8ccf-2095-47da-b94c-55fa1d0f2186">

**Comment by praymich** [comment]
你好，我想问下，以这种方式大概去微调三万多条中文的图文数据，能够让他有中文回复能力，并且学到里面的信息吗？
